packages/yew/src/functional/hooks/use_async.rs

```diff
@@ -0,0 +1,56 @@
+use std::{fmt, future::Future};
+
+use crate::{functional::hook, use_effect_with, use_state, UseStateHandle};
+
+#[hook]
```

Comment:
Ok so now I feel stupid. I had looked into `yew::functional` for hooks, and had totally missed `yew::suspense` also having a hook, that seems to be exactly what I was looking for…

So now I'm wondering, what value do you see in `use_async`, that would not be covered by `use_future` already? AFAICT it's possible to just match on the `SuspensionResult` instead of `?`-ing it out, so I'm not seeing much value right now, but maybe I'm missing something?

(Also, looking at the implementation of it resulted in [this PR](https://github.com/yewstack/yew/pull/3610), but that's unrelated)

Reply:
I can confirm that [I was able to replace my personal use of `use_async` with `use_future_with`](https://github.com/Ekleog/crdb/commit/b3f217b6b56ba247c58a2640f632ab2e25bcef1a). Maybe the main thing to improve about `use_future` would be re-exporting it from `yew::functional`, so that it's in the prelude, and maybe to document it a bit more prominently?

Reply:
The value would be that `use_async` requires knowledge of `Suspension` and this might not always be what the user intends to do. Note that the following code is *not* the equivalent of what I'd expect `use_async` to do:

```rust
let future: impl Future<Result = T> = todo!();
let foo = use_future(future).ok(); // An Option<T>, sucess?!
```

In fact, dropping the `Suspension` - instead of returning it from a component function - will subsequently not lead a re-render when the future resolves! `Suspension` and `use_future` want to say: this component is not ready to render yet, wait for a bit until I tell you to try again. `use_async` should say: Resolve this future, but I know what to do while the data is not there yet, keep on rendering.

Reply:
> I can confirm that [I was able to replace my personal use of `use_async` with `use_future_with`](https://github.com/Ekleog/crdb/commit/b3f217b6b56ba247c58a2640f632ab2e25bcef1a). Maybe the main thing to improve about `use_future` would be re-exporting it from `yew::functional`, so that it's in the prelude, and maybe to document it a bit more prominently?

That is a bit surprising to me, too. Do these components perhaps re-render because of some other thing happening instead of the Suspension resolving?

Reply:
Hmm is it supposed to not resolve? Looking at these lines:
- [this `output.set` will trigger a rerender, because the `use_state` value changed](https://docs.rs/yew/latest/src/yew/suspense/hooks.rs.html#112), and
- [`Suspension::from_future` will poll the future to completion even if dropped, so it will indeed be called at some point](https://docs.rs/yew/latest/src/yew/suspense/suspension.rs.html#61-70)

Is my new code relying on an unintended feature that could disappear in the next version? :sweat_smile: (Right now I don't care much because it's only example code, but I'm planning on writing a `crdb-yew` crate next that'll be basically implementing lots of custom async-using yew hooks, so I'd rather not make that mistake there too)